### ISSUE TITLE
Ability to populate subject field (C, O, OU etc) when generating certificates from PKI roles

### ISSUE THREAD

**Issue by alem117** [hashicorp/vault#9774]
Hello, we're building a PKI system using Vault, and we want to fill in the organizational unit (OU) field for each certificate manually. This feature is already available when generating root and intermediate certificates, but it's missing on end-entity certificates, which is what we need.

### Vault Version
Vault v1.4.2

### Requested feature
When generating a certificate from a role, we would like it to be similar to the following example
`vault write pki/issue/example_role common_name="ftp.example.com" organization="example.inc" ou="development team" country="EX"`

### Use case
Our organization has a lot of OU's, it would be a hassle to create a role for each one of them, we would like to only create one role that generates all the certificates we need, given that we can manually enter the ou value for each certificate request manually.

**Comment by pablo-ruth** [comment]
Hello,

We need this too, for managing Kubernetes client auth certificates. The organization fields are used to pass groups a user belongs to: https://kubernetes.io/docs/reference/access-authn-authz/authentication/#x509-client-certs
Since each user can have different combinations of groups, it means that we have to create thousands of roles.

**Comment by HridoyRoy** [comment]
Hi all,
Thanks for your patience. I'll take a look at this and circle back.

**Comment by amjanoni** [comment]
I thought that using CSR would respect the info from there but this is not what happens.

+1 for this feature

**Comment by Cyb3rYoga** [comment]
+1 for this feature, it would solve a lot of problems we're having.

**Comment by luisyonaldo** [comment]
Hi, we are also using Vault PKI to generate certificates to authenticate users in k8s, but we are blocked by this. The only way to be able to do this atm is on the `role` level, which mean we have to create a role per user and keep the `organization` in sync with LDAP.

Is this being worked? can I have a bit of guidance to help with a PR?

**Comment by cipherboy** [comment]
As a note, this is a duplicate of #4562. As we have multiple issues with the same topic (give or take CSR vs role-generated -- which will likely both be solved together), I'm closing this towards that one. See some context on why this is a little non-trivial: https://github.com/hashicorp/vault/issues/4562#issuecomment-1125413160
